service-impl/resource-service-impl/egeria-resource-service/src/main/java/uk/gov/gchq/palisade/resource/service/impl/EgeriaResourceService.java

```diff
@@ -0,0 +1,287 @@
+                    LOGGER.error(e.getMessage());
+                    return null;
+                }
+            })
+        .filter(Objects::nonNull);
+    }
+
+    Stream<AssetUniverse> tagWithMetadata(final Stream<AssetUniverse> assets, final UserId userId, final String comment) {
+        return assets.peek(asset -> {
+                try {
+                    assetConsumer.addCommentToAsset(userId.getId(), asset.getGUID(), CommentType.STANDARD_COMMENT, comment, true);
+                    LOGGER.debug("Added comment {} to asset {}", comment, asset.getGUID());
+                } catch (Throwable e) {
+                    LOGGER.warn("Exception while adding comment to asset {}:", asset.getGUID());
+                    LOGGER.warn(e.getMessage());
+                }
+            });
+    }
+
+    Stream<SimpleEntry<AssetUniverse, ConnectionDetail>> withConnectionDetails(final Stream<AssetUniverse> assets, final UserId userId) {
+        return assets.map(asset -> {
+                ConnectionDetail connectionDetail = null;
+                try {
+                    //connector = assetConsumer.getConnectorForAsset(userId.getId(), asset.getGUID());
+                    AdditionalProperties props = asset.getAdditionalProperties();
+                    Class<? extends ProxyRestService> serviceClass = (Class<? extends ProxyRestService>) Class.forName(props.getProperty("proxyRestServiceClass"));
+                    String url = props.getProperty("proxyRestServiceUrl");
+                    connectionDetail = new ProxyRestConnectionDetail().serviceClass(serviceClass).url(url);
+                } catch (Throwable e) {
+                    LOGGER.warn("Exception while getting connection details for asset {}:", asset.getGUID());
+                    LOGGER.warn(e.getMessage());
+                }
+                return new SimpleEntry<>(asset, connectionDetail);
+            });
+    }
+
+    static ChildResource resolveParents(final ChildResource resource) {
+        String path = resource.getId();
+        if (path.endsWith(File.separator)) {
+            path = path.substring(0, path.length() - 1);
+        }
+        String rootPath = path.substring(0, path.indexOf(File.separator));
+        String parentPath = path.substring(0, path.lastIndexOf(File.separator));
+        if (parentPath.equals(rootPath)) {
+            SystemResource root = new SystemResource().id(parentPath);
+            resource.setParent(root);
+        } else {
+            DirectoryResource parent = new DirectoryResource().id(parentPath);
+            resolveParents(parent);
+            resource.setParent(parent);
+        }
+        return resource;
+    }
+
+    CompletableFuture<Map<LeafResource, ConnectionDetail>> asResourceMap(final Stream<SimpleEntry<AssetUniverse, ConnectionDetail>> assetConnections) {
+        return CompletableFuture.supplyAsync(() -> assetConnections
+                //.filter(entry -> !entry.getKey().getAssetTypeName().equals("FileFolder"))
```

Comment:
I thought this filter was required

Reply:
It depends, i left it as 'if you request a directory, you get a directory'

Reply:
it should be if you request a directory, Palisade will give you all the resources in that directory